vernac/vernacentries.ml

```diff
@@ -882,16 +900,21 @@ let preprocess_inductive_decl ~atts kind indl =
     if List.for_all is_record indl then primitive_proj
     else Notations.return false
   in
-  let (((template, (poly, cumulative)), private_ind), typing_flags), primitive_proj =
+  let hint_mode_attr : Hints.hint_mode list option Attributes.attribute =
+    match kind with
+    | Class _ -> mode_attr
+    | _ -> Notations.return None
+  in
+  let ((((template, (poly, cumulative)), private_ind), typing_flags), primitive_proj), mode =
     Attributes.(
       parse Notations.(
           template
           ++ polymorphic_cumulative ~is_defclass:(Option.has_some is_defclass)
-          ++ private_ind ++ typing_flags ++ prim_proj_attr)
+          ++ private_ind ++ typing_flags ++ prim_proj_attr ++ hint_mode_attr)
```

Comment:
This attribute does not seem to be tested in the test suite and also it does not error (just ignore it) if the inductive is not a Class.

Reply:
Now tested. It does fail if the command is not a Class.